aptos-move/aptos-vm-types/src/change_set.rs

```diff
@@ -27,12 +33,12 @@ use std::collections::{
 /// VM. For storage backends, use `ChangeSet`.
 #[derive(Debug, Clone, Eq, PartialEq)]
 pub struct VMChangeSet {
-    resource_write_set: HashMap<StateKey, WriteOp>,
+    resource_write_set: HashMap<StateKey, (WriteOp, Option<MoveTypeLayout>)>,
     module_write_set: HashMap<StateKey, WriteOp>,
     aggregator_v1_write_set: HashMap<StateKey, WriteOp>,
     aggregator_v1_delta_set: HashMap<StateKey, DeltaOp>,
     aggregator_v2_change_set: HashMap<AggregatorID, AggregatorChange>,
-    events: Vec<ContractEvent>,
+    events: Vec<(ContractEvent, Arc<Option<MoveTypeLayout>>)>,
```

Comment:
Does the VM internally have an Arc, so that we don't create a copy at all? If not - having arc here doesn't make too much sense - we can create an Arc when storing it in MVHashMap

but also, we should have both resource_write_set and events be consistent - either both having Arc or none?

Also I think the other way around - `Option<Arc<MoveTypeLayout>>` makes more sense

Reply:
This should not be arc I think, change set owns all these values and they come from non-arced things in the VM. Having Arc in mvhashmap sounds good👍